execution_engine/src/storage/trie/mod.rs

```diff
@@ -526,16 +523,19 @@ impl<K, V> Trie<K, V> {
     }
 
     /// Returns the hash of this Trie.
-    pub fn hash(&self) -> Result<Digest, TrieHashingError>
+    pub fn trie_hash(&self) -> Result<Digest, bytesrepr::Error>
     where
         Self: ToBytes,
     {
         let bytes = self.to_bytes()?;
         if bytes.len() <= ChunkWithProof::CHUNK_SIZE_BYTES {
             Ok(Digest::hash(&bytes))
         } else {
-            let chunk_with_proof = ChunkWithProof::new(&bytes, 0)?;
-            Ok(chunk_with_proof.proof().root_hash())
+            Ok(Digest::hash_merkle_tree(
+                bytes
+                    .chunks(ChunkWithProof::CHUNK_SIZE_BYTES)
+                    .map(Digest::hash),
+            ))
```

Comment:
But that's not the same, is it? I think we still hash the result together with the chunk count?

Reply:
Yes, but `hash_merkle_tree` does that.
Also, the `returns_trie_or_chunk` test in `execution_engine` checks whether the hash of the trie is equal to the root hash of the chunks, and that passes, so the result _is_ the same.